flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetWriters.java

```diff
@@ -450,7 +450,10 @@ public boolean hasNext() {
 
     @Override
     protected Object get(RowData struct, int index) {
-      return fieldGetter[index].getFieldOrNull(struct);
+      if (struct.isNullAt(index)) {
+        return null;
+      }
```

Comment:
Why did this need to change? NPE in a test?

Reply:
There is an NPE in some test cases, yes.

Reply:
However, I'm going to investigate a bit furhter as I do think it might be indicative of a bug.

I think if we use the correct deletion schema in all cases, the NPEs will go away. I am testing now.

Reply:
Or actually, I don't think this change is needed. If there's no `fieldGetter` for a given index, that's likely indicative of a bug.

Reply:
Yeah this didn't need to be changed if we use the full schema as the deletion schema outside of `upsert` case like here https://github.com/apache/iceberg/pull/4364/files#diff-bbdfbcbd83d2e6f53d402e804dcd0a6fd28ab39d1cc3f74a88641ae8763fde3bR75-R87